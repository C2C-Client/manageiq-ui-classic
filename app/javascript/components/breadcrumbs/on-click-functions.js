export const onClickTree = (controllerName, item) => (
  window.miqCheckForChanges()
    ? sendDataWithRx({ breadcrumbSelect: { path: `/${controllerName}/tree_select`, key: item.key } }) : null
);

export const onClick = (e) => {
  if (!window.miqCheckForChanges()) {
    e.preventDefault();
  }
};
