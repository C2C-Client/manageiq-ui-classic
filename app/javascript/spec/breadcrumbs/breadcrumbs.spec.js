import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as clickFunctions from '../../components/breadcrumbs/on-click-functions';
import Breadcrumbs from '../../components/breadcrumbs';

describe('Breadcrumbs component', () => {
  const props = {
    items: [
      { url: '/providers', title: 'Providers' },
      { key: 'in tree', title: 'Google' },
      { action: 'accordion-select', title: 'All' },
      { title: 'This' },
    ],
    title: 'Title',
    controllerName: 'provider',
  };

  it('is correctly rendered', () => {
    const wrapper = mount(<Breadcrumbs {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('clicked on breadcrumb in tree', () => {
    clickFunctions.onClickTree = jest.fn();

    const initialProps = {
      ...props,
      items: [
        { key: 'xx-11', title: 'Item 11' },
        { title: 'Header' },
      ],
    };
    const wrapper = mount(<Breadcrumbs {...initialProps} />);

    wrapper.find('a').first().simulate('click');

    expect(clickFunctions.onClickTree).toHaveBeenCalledWith(initialProps.controllerName, initialProps.items[0]);
  });

  it('clicked on breadcrumb', () => {
    clickFunctions.onClick = jest.fn();

    const initialProps = {
      ...props,
      items: [
        { url: 'xx-11', title: 'Item 11' },
        { title: 'Header' },
      ],
    };
    const wrapper = mount(<Breadcrumbs {...initialProps} />);

    wrapper.find('a').first().simulate('click');

    expect(clickFunctions.onClick).toHaveBeenCalledWith(expect.any(Object), initialProps.items[0].url);
  });
});
