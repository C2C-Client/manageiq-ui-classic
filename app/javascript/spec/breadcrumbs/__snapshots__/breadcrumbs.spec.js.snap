// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component is correctly rendered 1`] = `
<Breadcrumbs
  controllerName="provider"
  items={
    Array [
      Object {
        "title": "Providers",
        "url": "/providers",
      },
      Object {
        "key": "in tree",
        "title": "Google",
      },
      Object {
        "action": "accordion-select",
        "title": "All",
      },
      Object {
        "title": "This",
      },
    ]
  }
  title="Title"
>
  <Breadcrumb
    style={
      Object {
        "marginBottom": 0,
      }
    }
    title={false}
  >
    <Breadcrumb
      bsClass="breadcrumb"
      className=""
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      <ol
        aria-label="breadcrumbs"
        className="breadcrumb"
        role="navigation"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <BreadcrumbItem
          active={false}
          href="/providers"
          key="/providers"
          onClick={[Function]}
        >
          <li
            className=""
          >
            <SafeAnchor
              componentClass="a"
              href="/providers"
              onClick={[Function]}
            >
              <a
                href="/providers"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Providers
              </a>
            </SafeAnchor>
          </li>
        </BreadcrumbItem>
        <BreadcrumbItem
          active={false}
          key="in tree-1"
          onClick={[Function]}
        >
          <li
            className=""
          >
            <SafeAnchor
              componentClass="a"
              onClick={[Function]}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                Google
              </a>
            </SafeAnchor>
          </li>
        </BreadcrumbItem>
        <li
          key="2"
        >
          All
        </li>
        <BreadcrumbItem
          active={true}
        >
          <li
            className="active"
          >
            <span>
              <strong>
                This
              </strong>
            </span>
          </li>
        </BreadcrumbItem>
      </ol>
    </Breadcrumb>
  </Breadcrumb>
</Breadcrumbs>
`;
